src/main/java/com/notably/model/suggestion/SuggestionItemImpl.java

```diff
@@ -42,4 +42,20 @@ public SuggestionItemImpl(String displayText, int frequency, Runnable action) {
     public Runnable getAction() {
         return action;
     }
+
+    @Override
+    public int compareTo(SuggestionItem item) {
+        Objects.requireNonNull(item);
+
+        if (getProperty("frequency").isPresent() && item.getProperty("frequency").isPresent()) {
+            int thisFrequency = Integer.parseInt(getProperty("frequency").get());
+            int otherFrequency = Integer.parseInt(item.getProperty("frequency").get());
+            if (thisFrequency != otherFrequency) {
+                return thisFrequency - otherFrequency;
+            }
+        }
+
+        return 0;
+    }
+
```

Comment:
Hmmm, I personally feels that this `compareTo` method is semantically incomplete since you're only considering the case when `"frequency"` is present. If it's not, you'll return `0`, which means two `SuggestionItem` is equal. The point is, this `compareTo` method is not general, IMO. It is only tuned for comparing between to search suggestion items.

If I were you, I'd rather not make `SuggestionItemImpl` comparable and instead move this sorting logic to the `Collections#sort` method. The `Collections#sort` method has a second variant in which there is a second `Comparator` parameter. You can then execute it as such:

```java
Collections.sort(suggestions, (suggestion1, suggestion2) -> {
  if (!suggestion1.getProperty("frequency").isPresent()
      || !suggestion2.getProperty("frequency").isPresent()) {
    throw new AssertionError("All search suggestion item must contain the \"frequency\" property");
  }
  if (!suggestion1.getProperty("displayText").isPresent()
      || !suggestion2.getProperty("displayText").isPresent()) {
    throw new AssertionError("All search suggestion item must contain the \"displayText\" property");
  }

  int frequency1 = Integer.parse(suggestion1.getProperty("frequency").get());
  int frequency2 = Integer.parse(suggestion2.getProperty("frequency").get());

  if (frequency1 == frequency2) {
    String displayText1 = suggestion1.getProperty("displayText").get();
    String displayText2 = suggestion2.getProperty("displayText").get();
    return displayText1.compareToIgnoreCase(displayText2);
  }

  return frequency1 - frequency2;
});
```

Reply:
Oh wow, didn't know this before. Thanks @kevinputera ! Will change!